source/cluster_manager/clusterManager.js

```diff
@@ -555,204 +566,945 @@ var ClusterManager = function (clusterName, selfId, spec) {
     return that;
 };
 
-var runAsSlave = function(topicChannel, manager) {
-    var loss_count = 0,
-        interval;
+var getTimerCount = function (sec, timerStep) {
+    if (timerStep <= 0) {
+        timerStep = 100;
+    }
+    if (timerStep > sec) {
+        sec = timerStep;
+    }
+    return parseInt((sec + timerStep - 1) / timerStep);
+}
 
-    clearRouterKeys(topicChannel,manager);
-    role = "slave";
+class HeartbeatObserver {
+    constructor(onLoss) {
+        this.interval = undefined;
+        this.lastContact = new Date();
+        this.state = state;
+        this.onLoss = onLoss;
+        this.intervalStep = 100;
+        this.mixStepCount = getTimerCount(parseInt(state.electTimeout/2), this.intervalStep);
```

Comment:
I'm not very familiar with raft, what `getTimerCount` and `stepCount` used for?
Does `mixStepCount` mean `minStepCount`?

Reply:
> I'm not very familiar with raft, what `getTimerCount` and `stepCount` used for? Does `mixStepCount` mean `minStepCount`?

yes, is minStepCount,i coding error let me fix it
The getTimerCount method simply converts time into the number of intervals based on the intervalStep timing cycle. Since the heartbeat timing cycle is 100ms, we set the intervalStep to 100ms. The mixStepCount and maxStepCount are calculated using getTimerCount based on the electTimeout and the intervalStep to determine how many cycles should be executed. As the Raft candidate uses a random election timeout to resolve multiple term conflicts, it is recommended to be between 150-300ms. Therefore, we set the mixStepCount to half of the maxStepCount according to this recommendation. The electTimeout for etcd is 1s, while I have configured it to be 4s for my own use